addons/web/static/src/core/utils/strings.js

```diff
@@ -10,15 +33,14 @@ export const nbsp = "\u00a0";
  * @returns {string} an escaped string
  */
 export function escape(str) {
-    if (str === undefined) {
-        return "";
-    }
-    if (typeof str === "number") {
-        return String(str);
+    if (typeof str === "object" && str[escapeMethod]) {
+        return str[escapeMethod]();
```

Comment:
There are different ways to escape for different use.
- escape html
- escape strings for usage in XML
- escape strings to use in js code
- escape strings for URL (encodeUriComponent)

I think we should have one explicit tool for each case, instead of a overarching one like underscore does.
What do you think @sdegueldre @ged-odoo @aab-odoo  ?

Reply:
I wouldn't say this is overreaching, this method only escapes for HTML/XML. Quotes need to be escaped in attributes and we don't know the caller's intention, I think it safer to just escape characters that need to be escaped in both content and attributes. This function doesn't cover the other cases.

The escapeMethod thing is for markup objects but this implementation is incorrect because it doesn't use the html symbol that's used by the markup object so it will escape even markup.

Reply:
> I wouldn't say this is overreaching, this method only escapes for HTML/XML. Quotes need to be escaped in attributes and we don't know the caller's intention, I think it safer to just escape characters that need to be escaped in both content and attributes. This function doesn't cover the other cases.

I agree

Reply:
export const escapeMethod = Symbol("html");
The above exported is used by escape() and by Markup() and then markup is not escaped when used in escape().
This case is tested in qweb_tests.js. 
I'm agree about the caller intention this is why I'll let the previous implementation.

Reply:
Not sure if the code was in a different state when I reviewed or I missed the fact that it was exported here and imported in markup but in this current state this seems good to me :+1: